From 6ed206a3f38a1bc5af315b2794d26392d45ba434 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Wed, 17 Feb 2010 11:40:31 +0100
Subject: [PATCH] avoid buffer overflow in gegl_buffer_header_init()

Use memcpy() instead of strcpy() as the target is not a zero-terminated
string.
---
 gegl/buffer/gegl-buffer-save.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gegl/buffer/gegl-buffer-save.c b/gegl/buffer/gegl-buffer-save.c
index 85c7894..aff175c 100644
--- a/gegl/buffer/gegl-buffer-save.c
+++ b/gegl/buffer/gegl-buffer-save.c
@@ -196,7 +196,7 @@ gegl_buffer_header_init (GeglBufferHeader *header,
                          gint              bpp,
                          Babl*             format)
 {
-  strcpy (header->magic, "GEGL");
+  memcpy (header->magic, "GEGL", 4);
 
   header->flags = GEGL_FLAG_HEADER;
   header->tile_width  = tile_width;
-- 
1.6.6

